tests/unit/_modules/test_packagemanager.py

```diff
@@ -0,0 +1,345 @@
+import pytest
+from srv.salt._modules.packagemanager import PackageManager, Zypper, Apt
+from mock import MagicMock, patch, mock
+
+
+class TestPackageManager():
+    '''
+    This class contains a set of functions that test srv.salt._modules.packagemanager
+    '''
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_opensuse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('opensuse', 42.1, 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_suse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('SUSE', '12.2', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_centos(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('centos', '8', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_fedora(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('fedora', '23', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_reboot(self, po, dist_return):
+        """
+        Reboot method
+        """
+        dist_return.return_value = ('opensuse', '42.2', 'x86_64')
+        ret = PackageManager()
+        ret.reboot_in()
+        assert po.called is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_not_implemented(self, po, dist_return):
+        """
+        Your platform is not supported
+        """
+        dist_return.return_value = ('ScientificLinux', '42.2', 'x86_64')
```

Comment:
> . I think this will be able to support more systems with less code/work and will support the corner case where a distro changes from one package manager to another as Fedora has done more easily than hard-coding versions. I suspect this would make this particular code need less future maintenance. Do you think this logic holds up?

Right, makes sense. But let's keep that in mind and rework that once the PR is in. (track that with an issue).